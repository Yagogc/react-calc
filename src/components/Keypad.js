import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Keypad = ({
  callOperator,
  numbers,
  operators,
  setOperator,
  updateDisplay
}) => {
  numbers = numbers.map(number => {
    return <p key={number}>{number}</p>;
  });
  operators = operators.map(operator => {
    return <p key={operator}>{operator}</p>;
  });
  return (
    <Container className="keypad-container">
      <Numbers className="numbers-container">{numbers}</Numbers>
      <Operators className="operators-container">{operators}</Operators>
    </Container>
  );
};

Keypad.propTypes = {
  callOperator: PropTypes.func.isRequired,
  numbers: PropTypes.array.isRequired,
  operators: PropTypes.array.isRequired,
  setOperator: PropTypes.func.isRequired,
  updateDisplay: PropTypes.func.isRequired
};

export default Keypad;

const Container = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  height: var(--keypad-height);
  padding: 2%;
  width: var(--keypad-width);
`;
Container.displayName = "div";

const Numbers = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  height: 80%;
  width: 75%;
`;
Numbers.displayName = "div";

const Operators = styled.div`
  display: flex;
  flex-direction: column;
  height: 80%;
  width: 25%;
`;
Operators.displayName = "div";

const Submit = styled.div`
  height: 20%;
  width: 100%;
`;
Submit.displayName = "div";
