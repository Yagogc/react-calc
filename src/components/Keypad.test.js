import React from "react";
import { shallow, mount } from "enzyme";
import Keypad from "./Keypad";

describe("Keypad", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Keypad
        callOperator={jest.fn()}
        numbers={[]}
        operators={[]}
        setOperator={jest.fn()}
        updateDisplay={jest.fn()}
      />
    );
  });
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should render a <div />", () => {
    expect(wrapper.find("div").length).toEqual(3);
  });
  it("should render an instance of the Key component for each index of numbers, operators, and the submit Key", () => {
    const numbers = ["0", "1"];
    const operators = ["+", "-"];
    const submit = 1;
    const keyTotal = numbers.length + operators.length + submit;
    wrapper.setProps({ numbers, operators });
    expect(wrapper.find("Key").length).toEqual(keyTotal);
  });
});

describe("mounted Keypad", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <Keypad
        callOperator={jest.fn()}
        numbers={[]}
        operators={[]}
        setOperator={jest.fn()}
        updateDisplay={jest.fn()}
      />
    );
  });

  /*   it("renders the values of numbers to the DOM", () => {
    wrapper.setProps({ numbers: ["0", "1", "2"] });
    console.log(wrapper.find(".numbers-container").hostNodes());
    wrapper.find(".numbers-container").forEach(node => {
      expect(node.text()).toEqual("012");
    });
  }); */
  it("renders the values of numbers to the DOM", () => {
    wrapper.setProps({ numbers: ["0", "1", "2"] });
    expect(
      wrapper
        .find(".numbers-container")
        .hostNodes()
        .text()
    ).toEqual("012");
  });

  it("renders the values of operators to the DOM", () => {
    wrapper.setProps({ operators: ["+", "-", "*", "/"] });
    expect(
      wrapper
        .find(".operators-container")
        .hostNodes()
        .text()
    ).toEqual("+-*/");
  });
});
