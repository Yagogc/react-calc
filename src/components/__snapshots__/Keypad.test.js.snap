// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Keypad
    callOperator={[Function]}
    numbers={Array []}
    operators={Array []}
    setOperator={[Function]}
    updateDisplay={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="numbers-container"
>
          
</div>,
        <div
          className="operators-container"
>
          
</div>,
        <Key
          keyAction={[Function]}
          keyType="submit-key"
          keyValue="="
/>,
      ],
      "className": "keypad-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [],
          "className": "numbers-container",
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [],
          "className": "operators-container",
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "keyAction": [Function],
          "keyType": "submit-key",
          "keyValue": "=",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="numbers-container"
>
            
</div>,
          <div
            className="operators-container"
>
            
</div>,
          <Key
            keyAction={[Function]}
            keyType="submit-key"
            keyValue="="
/>,
        ],
        "className": "keypad-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [],
            "className": "numbers-container",
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [],
            "className": "operators-container",
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "keyAction": [Function],
            "keyType": "submit-key",
            "keyValue": "=",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
